### PR TITLE
Nginx formplayer proxy

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride @snopoke (wasn't sure who to ping) this adds the ability to specify multiple balancers. essentially we'll be routing `/formplayer` requests to the formplayer server. i also did a pass to cleanup how locations were dealt with. Easiest read 🐟 . quite a lot of tech debt in this role

cc: @biyeun

### CODE REVIEW

File: ansible/roles/nginx/vars/cas_ssl.yml [hunk 1]
```diff
@@ -4,19 +4,22 @@ nginx_ssl_key:  "{{ cas_nginx_ssl_key }}"
...
-   balancer: webworkers
    proxy_set_headers:
    - "Host $http_host"
    - "X-Forwarded-For $remote_addr"
    add_header: "X-Frame-Options SAMEORIGIN"
    access_log: "{{ log_home }}/{{ deploy_env }}-timing.log timing"
    location1:
     name: /
-    balancer: webworkers
+    balancer: "{{ deploy_env }}_cas_commcare"
```
Comment (benrudolph) [inline ansible/roles/nginx/vars/cas_ssl.yml]: balancer now specifies the name of the balancer you'd like to use with that location

File: ansible/roles/nginx/vars/cchq_ssl.yml [hunk 2]
```diff
@@ -1,19 +1,25 @@
 ---
 nginx_sites:
 - server:
+   balancers:
+    - name: "{{ deploy_env }}_commcare"
+      hosts: webworkers
+      port: "{{ django_port }}"
+    - name: "{{ deploy_env }}_formplayer"
+      hosts: touchforms
+      port: 8181
```
Comment (benrudolph) [inline ansible/roles/nginx/vars/cchq_ssl.yml]: ...this should probably be a variable somewhere

**Comment by snopoke** [comment]
nice. Just to clarify - we're exposing formplayer directly to the web right? Does it have appropriate security etc?

**Comment by benrudolph** [comment]
that's correct. good question, not sure how to best approach that. @wpride maybe we can sit down and make sure that there aren't any obvious vulnerabilities? @snopoke do you have any suggestions? maybe there's an audit checklist somewhere

**Comment by wpride** [comment]
@benrudolph sounds good, will grab time

**Comment by snopoke** [comment]
My only current thought is to add some simple global auth to all endpoints
prevent spam requests: https://spring.io/guides/gs/securing-web/

**Comment by benrudolph** [comment]
ok this has been deployed to prod and staging, i think g2g

**Comment by benrudolph** [comment]
(doing india and swiss now)

**Comment by benrudolph** [comment]
deployed to all envs

**Comment by benrudolph** [comment]
ok finally deployed to every environment. there's one sticky wicket. both the `cas` and `cchq_ssl` both try to be the `default_server` which causes it to crash (this was an issue before my refactor)

**Comment by benrudolph** [comment]
✅

**Comment by snopoke** [comment]
not mergable

**Comment by benrudolph** [comment]
👍

**Comment by benrudolph** [comment]
(nvm this looks like it broke some motech configs)

**Comment by benrudolph** [comment]
nvm on that previous nvm. this is g2g

**Pull request merged**